DOCS-9403 Partner Manager Editing (#88)
mulesoft/docs-partner-manager: modules/ROOT/pages/activity-tracking-edit.adoc

= Activity Tracking

B2B transactions often communicate contractual obligations between companies. Due to the sensitivity of this information, it is desirable (and often necessary) to track transmissions and store both transmission content and metadata.

> Anypoint Partner Manager enables you to monitor trading partner transmission activity. Activity is the complete set of end-to-end events related to transmission metadata, messages, and errors that are involved in processing a transmission.

Reviewer: Just to emphasize/clarify in first sentence, although done in second more completely:
```suggestion
Anypoint Partner Manager enables you to monitor all trading partner transmission activity. _Activity_ is the complete set of end-to-end events related to transmission metadata, messages, and errors that are involved in processing a transmission.
```

> Metadata includes things like the date and time the transmission was sent or received, the identity of the partner, and the success or failure of processing.

Reviewer: Opt:
```suggestion
_Metadata_ includes things like the date and time the transmission was sent or received, the identity of the partner, and the success or failure of processing.
```
Is this success or failure separate from "errors" in the list above? If so, should the list above separate "errors" out, it that is metadata?

Common uses for this metadata include:

* Nonrepudiation
* Auditing
* Correlation of acknowledgments

> * Error reporting

Reviewer: Are these described somewhere?
I wonder if metadata isn't a topic unto itself....

> With activity tracking, you can:

Reviewer: I wonder if this text should be earlier in the intro? It seems a little after-the-fact to list uses of tracking after explaining all about metadata, which is part of the tracking but not all of it.

Also...I love the stuff about metadata; just not sure it belongs in this intro to tracking in general? We spend no time here on original payload or errors and acks. Should metadata have its own subtopic?

Reviewer: BTW...omg this is so much better! I can actually understand what is going on!!!!

> * Search for message metadata (such as partner name, message type, filename, control number, endpoint, or date received).

Reviewer: Repeats much of what is defined above for metadata, and adds to it; ditto re. consider separation of details into own topic.

* View and download the original payload related to the transmission.
* Review errors and acknowledgment payloads.

== Track Message Flow Activity

> To track message flow activity:

Reviewer: This really doesn't add any value, because it repeats the title directly after the title.
The optimal setup, IMHO, is the title, a brief statement giving a chronological context to the task--maybe there are some prereqs or something you should do first, or a certain time when you want to perform this task?--and then the Steps subhead.
But this might all be for another day...I know you are operating under special circumstances, and I should prolly just cool it with the org. ideas....

. Open Anypoint Partner Manager.
. Select *Activity* from the left navigation menu.

> . Locate a transmission you are interested in viewing. 

Reviewer: Why/under what circumstances would the audience want or need to view a transmission?
Maybe that needs adding to the intro?

> . Use the search box and filters to find the transmission based on criteria such as:

Reviewer: This is not a separate step; this is HOW you locate the transmission, so it should be a plain text sentence following the step.

+
* Trading partner name
* Message type
* Filename
* Control number
* Endpoint
* Date the file was received
. Select the transmission that matches your search criteria and review the summary information. +

> The summary information contains:

Reviewer: This c/b omitted by ending prev. sentence with a colon and going straight to list....

> * The status of the transmission, which indicates either a success (Green checkmark) or failure (red circle with an exclamation point)

Reviewer: ```suggestion
* The status of the transmission, which indicates either a success (green checkmark) or failure (red circle with an exclamation point)
```

* The name of the file that was received
* The control number associated with the file (if available)

> * The receiving endpoint at which the file was received

Reviewer: ```suggestion
* The endpoint at which the file was received
```

> * The date and time when the file was received

Reviewer: ```suggestion
* The date and time that the file was received
```

* The types of messages received (depending on the message flows you configured)
* The number of messages received in the file
+

> If there is an exclamation point (!) next to any of the rectangle boxes in the *Transmission Flow* section, this indicates there is an issue with the transmission.

Reviewer: ```suggestion
An exclamation point (!) next to any of the rectangle boxes in the *Transmission Flow* section indicates that there is an issue with the transmission.
```

Reviewer: But...I realize that the narrative flow is interrupted here.
We are listing the summary information in a set of steps. That list somewhat interrupts the step flow, and the information about the *Transmission Flow* section interrupts the list of summary information...:

. Locate
. Select and review:
*status
*name
...
*number
*TRANSMISSION FLOW

At this point ^, you can choose whether or not you really need to go in to detail about the section.
You could summarize it, something like this:

* Transmission flow, including processing steps and any issues

This assumes that the UI is fairly intuitive and that we don't need to describe what is shown onscreen.

If you keep the original text, I suggest ...:

* Transmission flow
   The *Transmission Flow* section of the summary information shows things like:
    * Success or failure (!)
    * Acknowledgement
    * Transformation and routing
. Click *View Payload* to view and download the original payload if you need to troubleshoot a failure.

The *Transmission Flow* section shows the steps the file was processed in, for example, if:
* The file was successfully received and routed
* An acknowledgement was sent back to the trading partner
* The message was appropriately transformed and routed to the target endpoint 
. Click *View Payload* to view and download the original payload that was received. +

> Examining the received payload can help you to determine where there might be a problem in the message flow.

Reviewer: See above
